packages/hmr/src/index.ts

```diff
@@ -0,0 +1,57 @@
+import { container, Piece, Store } from '@sapphire/framework';
+import { watch } from 'chokidar';
+import { relative } from 'path';
+
+/**
+ * Starts HMR for all registered {@link Store Stores} in {@link container.stores the main container}.
+ */
+export function start() {
+	container.logger.info('HMR is enabled!');
+	for (const store of container.stores.values()) {
+		const deleteCb = handlePiecePathDelete.bind(null, store);
+		const updateCb = handlePiecePathUpdate.bind(null, store);
+
+		watch([...store.paths])
+			.on('change', updateCb)
+			.on('unlink', deleteCb);
+	}
+}
+
+async function handlePiecePathDelete(store: Store<Piece>, path: string) {
+	if (!store.strategy.filter(path)) return;
+
+	const pieceToDelete = store.find((piece) => piece.location.full === path);
+	if (!pieceToDelete) return;
+
+	try {
+		await pieceToDelete.unload();
+		container.logger.info(`[HMR] Unloaded ${pieceToDelete.name}`);
+	} catch (error) {
+		container.logger.error(`[HMR] Failed to unload ${pieceToDelete.name}`, error);
+	}
```

Comment:
Seeing as this package depends on framework anyway we might as well use `fromAsync` here

```suggestion
const result = fromAsync(async () => {
	await pieceToDelete.unload();
	container.logger.info(`[HMR] Unloaded ${pieceToDelete.name}`);
});

if (isErr(result)) {
	container.logger.error(`[HMR] Failed to unload ${pieceToDelete.name}`, error);
}
```